packages/core/src/L1/synthesizers/FileSynthesizer.ts

```diff
@@ -165,35 +166,31 @@ export class FileSynthesizer extends Construct {
       return fs.readFileSync(realPath);
     }
 
-    return undefined;
+    return;
   }
 
   public readRealFile(who: Construct, filePath: string) {
     const data = this.tryReadRealFile(who, filePath);
 
-    if (!data) {
-      throw new Error(`Cannot read file at: ${filePath}`);
-    }
+    assert(Buffer.isBuffer(data), `Cannot read file at: ${filePath}`);
 
     return data;
   }
 
   public tryReadRealJsonFile<T = Record<string, unknown>>(who: Construct, filePath: string): T | undefined {
-    const data = this.tryReadRealFile(who, filePath) as Buffer;
+    const data = this.tryReadRealFile(who, filePath);
 
-    if (data) {
```

Comment:
This would fail if the file existed but was blank (zero bytes).

Reply:
Good find